### PR TITLE
[LT] Aftermath Support: Fixed Custom Flags, Fixed Civilian Template Equipment Loading, Undercover Adjustments

### LINKED ISSUE #724: HQ flag texture is never set
Some players would like to play with custom flags, changing the flag texture however is not possible, as the HQ flag texture is never set. Change that so players can use custom flags.

### PR DESCRIPTION
**Pull request by Iamgoofball**
## What type of PR is this.
1. [x] Bug
2. [x] Change
3. [x] Enhancement

### What have you changed and why?
Information:
Added support for the Aftermath mod to Antistasi's template system.
Fixed custom flags not working for Templates.
Fixed Civilian Template uniforms/headgear not being utilized.
Makes all gear defined in the Civilian Template count for Undercover
Makes all civilian gear count for Undercover.

Aftermath activates if Aftermath and Unsung are detected. Unsung vehicles are utilized due to a lack of themed vehicles, but the cold war tech fits well with the Aftermath universe equipment and feel.
Workshop collection utilized located here:
https://steamcommunity.com/workshop/filedetails/?id=2410758646

### Please specify which Issue this PR Resolves.
Fixes #724

### Please verify the following and ensure all checks are completed.

1. [X] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [ ] No
2. [X] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:

Todo:
Aftermath specific:
- [x] Base faction defines(Khans, Caesar's, NCR)
- [ ] Flesh out weaponry for all 3 factions
- [ ] Flesh out civilian vehicles
Antistasi specific:
- [ ] Set up civilian clothing defines for other templates since they need them now
- [ ] Add support for hidden holstered weapons not breaking concealment (IE. Holdout weapons that are intended to be concealed)

## Screenshots
steamwebhelper_WVGWNjoHON

### CODE REVIEW

**Comment by Bob-Murphy** [comment]
Hi Goofball,

me and some devs took a look at your PR and whilst the intention certainly is good, the implementation unfortunately leaves a lot to be desired.
Here a few things that went wrong:
- naming conventions of files were not followed as should - spaces and special characters might break shit
- templates are not sufficiently filled and therefore wouldn't work. Example: no weapons are properly defined in `Great Khans.sqf` loadout section
- you put template changes as well as not-really-necessary bugfixes (the flag stuff) into one PR - completely OOS (out of scope)
- breaks other modsets due to the changes in selector.sqf - which is a complete NO-GO
- re-enabled forbidden uniforms in fn_equipmentSort.sqf - which is a complete NO-GO
- changed gear which can be used for undercover - which is a complete NO-GO

Due to the listed issues as well as the fact that we have not even fully completed the backbone of the new template system, I will close this PR.
We most likely will utilize parts of this PR later on when/if we are enabling support for other mods, but this PR would need more work in fixing it then doing it ourselves.

Thank you for your effort but unfortunately this is not in a usable state for us.
Best regards,
Bob

**Comment by Iamgoofball** [comment]
> naming conventions of files were not followed as should - spaces and special characters might break shit

I've tested this and it seems to work just fine in game. Will need to test w/ linux.

>templates are not sufficiently filled and therefore wouldn't work. Example: no weapons are properly defined in Great Khans.sqf loadout section

The RebelExample.sqf that I modified to create Great Khans.sqf specified not to modify anything in the loadout section.
https://github.com/official-antistasi-community/A3-Antistasi/blob/loadout_templates/A3-Antistasi/Templates/NewTemplates/%23Examples/RebelExample.sqf#L60
As you can see here, I was complying with the listed documentation in the example files. You may want to update the example files if template authors are supposed to actually fill that out.

>you put template changes as well as not-really-necessary bugfixes (the flag stuff) into one PR - completely OOS (out of scope)

The only template changes in this are to my own templates. The bugfixes are bugfixes to features I utilize in this PR, however, I could atomize those into a separate PR for you.

>breaks other modsets due to the changes in selector.sqf - which is a complete NO-GO

Can you verify this claim? I see no reason that any other mods would be broken by this pull request, seeing as the formatting in selector.sqf. Which mods in question are broken by the changes to selector.sqf?

>re-enabled forbidden uniforms in fn_equipmentSort.sqf - which is a complete NO-GO

Those uniforms weren't actually re-enabled.

Basically, I switched the CivilianUniforms/CivilianHeadgear lists to be a whitelist determined by the loaded Civilian Template, instead of being "all uniforms in the civ pool".

This prevents issues with uniforms with incorrect tagging from showing up in the Arsenal and on Civilians(IE. in the Aftermath mod, there's some power armor pieces that are incorrectly tagged as Civilian), and means you can remove the massive hardcoded blacklist of forbidden uniforms from the code.

>changed gear which can be used for undercover - which is a complete NO-GO

See the above reasoning for the civilian change. This was a consistency fix to ensure that all the gear you see on civilians still counts for undercover. to avoid inconsistency issues.
